utils/src/tempfile.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::env::temp_dir;
+use std::path::Path;
+
+use vmm_sys_util::{errno::Result, tempfile};
+
+/// Wrapper over vmm_sys_util::tempfile::TempFile::new().
+/// This creates a temporary file inside $TMPDIR if set,
+/// otherwise inside `/tmp` if non-Android system.
+pub fn tempfile() -> Result<tempfile::TempFile> {
```

Comment:
I was picturing this more like a new Tempfile structure that extends the existing Tempfile from vmm-sys-util. I also thing that the goal should be to merge this changes in vmm-sys-util. To unblock ourselves it's okay to have these changes in Firecracker.

I quickly typed something that works as an example (it can be made nicer):
```rust
pub struct TempFile(tempfile::TempFile);

impl TempFile {
    fn new() -> Result<Self> {
        let mut in_tmp_dir = temp_dir();
        in_tmp_dir.push("");
        let temp_file = tempfile::TempFile::new(in_tmp_dir)?;
        Ok(TempFile(temp_file))
    }

    fn new_with_path(path: &Path) -> Result<Self> {
        let mut in_tmp_dir = temp_dir();
        let temp_file = tempfile::TempFile::new(in_tmp_dir)?;
        Ok(TempFile(temp_file))
    }

    fn path_ref(&self) -> &Path {
        self.0.as_path()
    }
}
```

The reason for which I prefer a structure is because I don't particularly like orphan functions, but this is subjective. Let me know what you think.

Reply:
Update: you don't need path_ref as you can implement deref for TempFile.

```
impl Deref for TempFile {
    type Target = tempfile::TempFile;

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
```

Reply:
Sure, this would work better, since it resembles the struct from `vmm-sys-utils`. I'll make it so.